python/pyarrow/tests/test_dataset.py

```diff
@@ -2582,6 +2600,38 @@ def test_open_dataset_from_fsspec(tempdir):
     assert dataset.schema.equals(table.schema)
 
 
+@pytest.mark.parquet
+@pytest.mark.s3
+def test_file_format_inspect_fsspec(s3_filesystem):
+    # https://issues.apache.org/jira/browse/ARROW-16413
+    from pyarrow.fs import _ensure_filesystem
+
+    fs, (host, port, access_key, secret_key) = s3_filesystem
+
+    # create bucket + file with pyarrow
+    table = pa.table({'a': [1, 2, 3]})
+    path = "mybucket/data.parquet"
+    with fs.open_output_stream(path) as out:
+        pq.write_table(table, out)
+
+    # read using fsspec filesystem
+    import s3fs
```

Comment:
Does this test get skipped if `s3fs` is not installed?

Reply:
Good catch, need to add some skips for this